src/server/controllers/user.js

```diff
@@ -6,31 +6,53 @@ const prisma = new PrismaClient();
 const jwtSecret = 'mysecret';
 
 const register = async (req, res) => {
-    const { username, password } = req.body;
-
-    const createdUser = null;
-
-    res.json({ data: createdUser });
+  const { username, password } = req.body;
+  // hash the password before sending to the database.
+  const hashed_password = await bcrypt.hash(password, 12);
+  try {
+    const createdUser = await prisma.user.create({
+      data: {
+        username,
+        password: hashed_password,
+      },
+    });
+    // delete password so it's not contained in the response to client.
+    delete createdUser.password;
+
+    // id and username are sent back to the user as a confirmation of success.
+    res.status(201).json({ data: createdUser });
+  } catch (error) {
+    return res
+      .status(400)
+      .json({ message: "Error creating user: " + error.message });
+  }
 };
 
 const login = async (req, res) => {
-    const { username, password } = req.body;
+  const { username, password } = req.body;
 
-    const foundUser = null;
+  try {
+    const foundUser = await prisma.user.findUnique({
+      where: { username: username },
```

Comment:
Very tiny thing, but instead of writing { username: username }, you can just write { username }, but this is more personal preference and the way you did it gives more visibility, so great job!